lib/src/linting/rules/has-discriminator.spec.ts

```diff
@@ -79,6 +80,63 @@ describe("rule: has-discriminator", () => {
     expect(errors).toEqual([]);
   });
 
+  test("valid for string literal unions", () => {
+    const errors = hasDiscriminator({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [],
+      types: [
+        {
+          name: "StringLiteralsUnion",
+          type: unionType([stringLiteral("a"), stringLiteral("b")])
+        }
+      ]
+    });
+    expect(errors).toEqual([]);
+  });
+
+  test("valid for nullable objects", () => {
+    const errors = hasDiscriminator({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [],
+      types: [
+        {
+          name: "NullableObject",
+          type: unionType([
+            objectType([
+              {
+                name: "name",
+                optional: false,
+                type: STRING
+              }
+            ]),
+            NULL
```

Comment:
ah interesting, does this mean that the generator will only generate 1 one-of child?

Reply:
No, the generator will directly generate the type with the `nullable` property set to true. So it's not generating a one-of at all :)